### ISSUE TITLE
Erroneous no-unused-vars exception when using variable as object key in destructuring assignment

### ISSUE THREAD

**Issue by jedverity** [eslint/eslint#2671]
This throws the no-unused-vars exception for `filename`:

```javascript
let filename = getFileName( foo, bar );
let { files: { [filename]: { url } } } = data;
```

I'll submit a patch if others agree this is not the desired behavior.

**Comment by AsaAyers** [comment]
if a variable is unused you can remove it without affecting the rest of the code. Clearly if you remove `filename` the code isn't going to work. I can't imagine an argument where that would be the desired behavior given your example.

**Comment by mysticatea** [comment]
I guess it had been fixed at #2635.

.eslintrc

```json
{
    "rules": {
        "no-unused-vars": 2
    },
    "ecmaFeatures": {
        "blockBindings": true,
        "destructuring": true,
        "objectLiteralComputedProperties": true
    }
}
```

test.js 

```js
let filename = getFileName( foo, bar );
let { files: { [filename]: { url } } } = data;
```

command

```
> eslint -v
v0.22.1
> eslint test.js --reset

test.js
  2:29  error  url is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by jedverity** [comment]
Ok great thx @AsaAyers @mysticatea
